account_asset_management/models/account_asset_line.py

```diff
@@ -0,0 +1,279 @@
+        store=True)
+    line_date = fields.Date(string='Date', required=True)
+    move_id = fields.Many2one(
+        comodel_name='account.move',
+        string='Depreciation Entry', readonly=True)
+    move_check = fields.Boolean(
+        compute='_compute_move_check',
+        string='Posted',
+        store=True)
+    type = fields.Selection(
+        selection=[
+            ('create', 'Depreciation Base'),
+            ('depreciate', 'Depreciation'),
+            ('remove', 'Asset Removal')],
+        readonly=True, default='depreciate')
+    init_entry = fields.Boolean(
+        string='Initial Balance Entry',
+        help="Set this flag for entries of previous fiscal years "
+             "for which OpenERP has not generated accounting entries.")
+
+    @api.depends('amount', 'previous_id', 'type')
+    @api.multi
+    def _compute_values(self):
+        dlines = self
+        if self.env.context.get('no_compute_asset_line_ids'):
+            # skip compute for lines in unlink
+            exclude_ids = self.env.context['no_compute_asset_line_ids']
+            dlines = dlines.filtered(lambda l: l.id not in exclude_ids)
+        dlines = self.filtered(lambda l: l.type == 'depreciate')
+        dlines = dlines.sorted(key=lambda l: l.line_date)
+
+        for i, dl in enumerate(dlines):
+            if i == 0:
+                depreciation_base = dl.depreciation_base
+                depreciated_value = dl.previous_id \
+                    and (depreciation_base - dl.previous_id.remaining_value) \
+                    or 0.0
+                remaining_value = \
+                    depreciation_base - depreciated_value - dl.amount
+            else:
+                depreciated_value += dl.previous_id.amount
+                remaining_value -= dl.amount
+            dl.depreciated_value = depreciated_value
+            dl.remaining_value = remaining_value
+
+    @api.depends('move_id')
+    @api.multi
+    def _compute_move_check(self):
+        for line in self:
+            line.move_check = bool(line.move_id)
+
+    @api.onchange('amount')
+    def _onchange_amount(self):
+        if self.type == 'depreciate':
+            self.remaining_value = self.depreciation_base - \
+                self.depreciated_value - self.amount
+
+    @api.multi
+    def write(self, vals):
+        for dl in self:
+            if vals.get('line_date'):
+                if isinstance(vals['line_date'], datetime.date):
+                    vals['line_date'] = fields.Date.to_string(
+                        vals['line_date'])
+            line_date = vals.get('line_date') or dl.line_date
+            asset_lines = dl.asset_id.depreciation_line_ids
+            if list(vals.keys()) == ['move_id'] and not vals['move_id']:
+                # allow to remove an accounting entry via the
+                # 'Delete Move' button on the depreciation lines.
+                if not self._context.get('unlink_from_asset'):
```

Comment:
Same here. `_context` is deprecated and would be best to be changed to `self.env.context`